processes/gops/process_info_linux.go

```diff
@@ -0,0 +1,20 @@
+package gops
+
+import (
+	"strings"
+
+	"github.com/shirou/gopsutil/process"
+)
+
+func pickName(p *process.Process) (string, error) {
+	cmdline, err := p.Cmdline()
+	if err != nil {
+		return "", err
+	}
+	// Assume that the process is a kernel process when it has no args
+	if strings.TrimSpace(cmdline) == "" {
```

Comment:
OK - that's weird, I have a bunch of processes in my personal-chef running with no arguments and no square brackets. Wondering if that description is a bit outtdated. Anyway, the square brackets are not there so that wouldn't cut it. So here is another suggestion (although we'd have to try it with some "older" distros as well): 

Why don't you mark as kernel processes those processes that are a descendent of `[kthreadd]`? I think this kernel process normally has a PID of `2`. That should be failsafe as `kthreadd` can only be your ancestor if you're a kernel process.